builds/flu/flu_titers.py

```diff
@@ -56,7 +56,122 @@ def HI_export(process):
+        else:
+            node.attr['age'] = 'unknown'
+        node.tip_ages = [] if node.attr['age']=="unknown" else [node.attr['age']]
+
+    # assign a list of ages to each internal node (will be used to calculate changes in age distribution)
+    for node in tree.get_nonterminals(order='postorder'):
+        valid_ages = [c.attr['age'] for c in node if c.attr['age'] is not "unknown"]
+        node.tip_ages = sum( [c.tip_ages for c in node] , [])
+        valid_tipcounts = [c.tip_count for c in node if c.attr['age'] is not "unknown"]
+        node.tip_count = np.sum(valid_tipcounts)
+        if len(valid_ages):
+            node.attr['age'] = np.sum([a*t for a,t in zip(valid_ages, valid_tipcounts)])/np.sum(valid_tipcounts)
+        else:
+            node.attr['age'] = "unknown"
+    root.outgroup_ages = []
+
+    # assign avg_age. This is redundant in that deep nodes have average ages already
+    # but terminal nodes or small clades don't have a smooth average. this assigns those
+    # to the parent.
+    # outgroup_ages: these are ages of all nodes NOT within the clade.
+    root.attr['avg_age'] = root.attr['age']
+    for node in tree.get_nonterminals(order='preorder'):
+        for c1 in node:
+            c1.outgroup_ages = node.outgroup_ages + sum([c2.tip_ages for c2 in node if c1!=c2], [])
+            if c1.tip_count<20:
+                c1.attr['avg_age'] = node.attr['avg_age']
+            else:
+                c1.attr['avg_age'] = c1.attr['age']
+
+    # calculate the a score to spot changing age distributions (for what ever reason, i.e. geography)
+    # these are calculated as a 2-sample KS test based on ages within and outside the clade in question
+    from scipy import stats
+    for node in tree.find_clades(order='preorder'):
+        if node==root:
+            node.attr['age_score'] = 0
+            continue
+        if len(node.tip_ages)>20 and len(node.outgroup_ages)>20:
+            ks = stats.ks_2samp(node.outgroup_ages, node.tip_ages)
+            node.attr['age_score'] = -np.log10(ks.pvalue)
+        else:
+            node.attr['age_score'] = node.up.attr['age_score']
+
+        if np.isnan(node.attr['age_score']):
+            node.attr['age_score']=0.0
+
+    # gender seems not to be a relevant quantity as of now...
+    for node in tree.get_terminals():
+        node.tip_count=1.0
+        if ('gender' in node.attr) and type(node.attr['gender']) in [str, unicode]:
+            g = node.attr["gender"]
+            node.attr["num_gender"] = -1 if g=='male' else (1 if g=='female' else 0)
+        else:
+            node.attr["num_gender"] = 0
+
+    for node in tree.get_nonterminals(order='postorder'):
+        node.tip_count = np.sum([c.tip_count for c in node])
+        node.attr['num_gender'] = np.sum([c.attr['num_gender']*c.tip_count for c in node])/node.tip_count
+
+
+    # not sure those have any effects right now... but auspice 2.0 might use them
+    runner.config["auspice"]["color_options"]["glyc"] = {
```

Comment:
@rneher: Shouldn't this be in `flu.process.py` and not in `flu_titers.py` along with other `runner.config` options?

Reply:
I see... this fits with placement of `ep`, `ne` and `rb`. Seems generally a little weird to have `H3N2_scores` as part of `flu_titers` to me.